core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorPageSourceProvider.java

```diff
@@ -28,4 +30,31 @@ ConnectorPageSource createPageSource(
             ConnectorTableHandle table,
             List<ColumnHandle> columns,
             DynamicFilter dynamicFilter);
+
+    /**
+     * Returns unenforced (effective) predicate that {@link ConnectorPageSource} would use to filter split data.
+     * If split is completely filtered out, then this method should return {@link TupleDomain#none}.
```

Comment:
> Unenforced predicate (or residual, or remaining predicate) typically refers to the part of the predicate that the connector is unable to guarantee.

This method is about `unenforced` predicates indeed.

> Effective predicate refers to the constraint data produce by a connector or subplan is guaranteed to satisfy.

In case of Hive `io.trino.plugin.hive.HiveTableHandle#compactEffectivePredicate` was always `effective`. I think the convention is used other place too (`io.trino.plugin.iceberg.IcebergPageSourceProvider#createPageSource`). Something we can change I guess. For now I've removed effective from javadoc